### PR TITLE
added watch and ownership for vmagent additional scrape config

### PR DESCRIPTION
**Pull request by f41gh7**
fixes to bug #19 
 Now changes to additional scrape config secret will trigger reconcile loop for vmagent.

### CODE REVIEW

File: pkg/controller/factory/vmagent.go [hunk 1]
```diff
@@ -485,3 +488,39 @@ func selectorLabelsVmAgent(cr *victoriametricsv1beta1.VmAgent) map[string]string
 	return labels
 
 }
+
+//add ownership - it needs for object changing tracking
+func addAddtionalScrapeConfigOwnership(cr *victoriametricsv1beta1.VmAgent, rclient client.Client, l logr.Logger) error {
+	if cr.Spec.AdditionalScrapeConfigs != nil {
```
Comment (tenmozes) [inline pkg/controller/factory/vmagent.go]: one of the go codestyle paradigm is to "return fast", so please change to 

`if cr.Spec.AdditionalScrapeConfigs != nil {
  return nil
}
`

File: pkg/controller/factory/vmagent.go [hunk 2]
```diff
@@ -485,3 +488,39 @@ func selectorLabelsVmAgent(cr *victoriametricsv1beta1.VmAgent) map[string]string
...
+		secret := &corev1.Secret{}
+		err := rclient.Get(context.TODO(), types.NamespacedName{Namespace: cr.Namespace, Name: cr.Spec.AdditionalScrapeConfigs.Name}, secret)
```
Comment (tenmozes) [inline pkg/controller/factory/vmagent.go]: don't need to use TODO context it just doesn't work, please pass it to function input or use context.Background

File: pkg/controller/factory/vmagent.go [hunk 3]
```diff
@@ -485,3 +488,39 @@ func selectorLabelsVmAgent(cr *victoriametricsv1beta1.VmAgent) map[string]string
...
+		if err != nil {
+			l.Error(err, "cannot find secret with scrape config", "secret", cr.Spec.AdditionalScrapeConfigs.Name)
```
Comment (tenmozes) [inline pkg/controller/factory/vmagent.go]: "secret <name> not found"

File: pkg/controller/factory/vmagent.go (hunk 1 above)
Comment (f41gh7) [reply pkg/controller/factory/vmagent.go]: changed with return fast.

File: pkg/controller/factory/vmagent.go (hunk 2 above)
Comment (f41gh7) [reply pkg/controller/factory/vmagent.go]: added background context.

File: pkg/controller/factory/vmagent.go (hunk 3 above)
Comment (f41gh7) [reply pkg/controller/factory/vmagent.go]: changed

**Pull request merged**